lib/http-bridge/test/integration/Test/Integration/Scenario/CLI/Transactions.hs

```diff
@@ -9,49 +11,427 @@ module Test.Integration.Scenario.CLI.Transactions
+        Stdout gOutDest <- getWalletViaCLI (T.unpack (wDest ^. walletId))
+        destJson <- expectValidJSON (Proxy @ApiWallet) gOutDest
+        verify destJson
+            [ expectCliFieldEqual balanceAvailable (2*amt)
+            , expectCliFieldEqual balanceTotal (2*amt)
+            ]
+
+    it "TRANS_CREATE_02 - Multiple Output Tx to different wallets via CLI" $ \ctx -> do
+        wSrc <- fixtureWallet ctx
+        wDest1 <- emptyWallet ctx
+        wDest2 <- emptyWallet ctx
+        addr1:_ <- listAddresses ctx wDest1
+        addr2:_ <- listAddresses ctx wDest2
+        let addr1' =
+                encodeAddress (Proxy @HttpBridge) (getApiT $ fst $ addr1 ^. #id)
+        let addr2' =
+                encodeAddress (Proxy @HttpBridge) (getApiT $ fst $ addr2 ^. #id)
+        let amt = 14
+        let (feeMin, feeMax) = (181487, 181839)
+        let args = T.unpack <$>
+                [ wSrc ^. walletId
+                , "--payment", T.pack (show amt) <> "@" <> addr1'
+                , "--payment", T.pack (show amt) <> "@" <> addr2'
+                ]
+
+        -- post transaction
+        (c, out, err) <- postTransactionViaCLI "cardano-wallet" args
+        err `shouldBe` "Please enter a passphrase: **************\nOk.\n"
+        txJson <- expectValidJSON (Proxy @(ApiTransaction HttpBridge)) out
+        verify txJson
+            [ expectCliFieldBetween amount (feeMin + amt, feeMax + amt)
+            , expectCliFieldEqual direction Outgoing
+            , expectCliFieldEqual status Pending
+            ]
+        c `shouldBe` ExitSuccess
+
+        -- verify balance on src wallet
+        Stdout gOutSrc <- getWalletViaCLI (T.unpack (wSrc ^. walletId))
+        gJson <- expectValidJSON (Proxy @ApiWallet) gOutSrc
+        verify gJson
+            [ expectCliFieldBetween balanceTotal
+                ( faucetAmt - feeMax - amt
+                , faucetAmt - feeMin - amt
+                )
+            , expectCliFieldEqual balanceAvailable (faucetAmt - 2*faucetUtxoAmt)
+            ]
+
+        forM_ [wDest1, wDest2] $ \wDest -> do
+            expectEventually' ctx balanceAvailable amt wDest
+            expectEventually' ctx balanceTotal amt wDest
+
+            -- verify balance on dest wallets
+            Stdout gOutDest <- getWalletViaCLI (T.unpack (wDest ^. walletId))
+            destJson <- expectValidJSON (Proxy @ApiWallet) gOutDest
+            verify destJson
+                [ expectCliFieldEqual balanceAvailable amt
+                , expectCliFieldEqual balanceTotal amt
+                ]
+
+    it "TRANS_CREATE_02 - Multiple Output Txs don't work on single UTxO" $ \ctx -> do
+        wSrc <- fixtureWalletWith ctx [2_124_333]
+        wDest <- emptyWallet ctx
+        addrs <- listAddresses ctx wDest
+
+        let addr1 =
+                encodeAddress (Proxy @HttpBridge) (getApiT $ fst $ addrs !! 1 ^. #id)
+        let addr2 =
+                encodeAddress (Proxy @HttpBridge) (getApiT $ fst $ addrs !! 2 ^. #id)
```

Comment:
Note that the target backend is now attached to the `Context` passed to the `Spec`. So we have in practice `Context t`. Alongside the `ScopedTypeVariable` language extension, you should now specify the target using `@t` instead of `@HttpBridge`. This enables the test scenario to be relatively backend-agnostic; such that we can soon enough replay them on top of Jormungandr without much efforts.

The choice of the target backend happens in the spec Main file. For each scenario, backend capabilities are specified as class constrained (you probably needn't bothering about that, GHC wil simply yell at you if you're missing one). 

For an example, see: https://github.com/input-output-hk/cardano-wallet/blob/d7e6fd68f2bbad6d1b47c5bc1b5372b9c7e2f72d/lib/http-bridge/test/integration/Test/Integration/Scenario/CLI/Addresses.hs#L38-L45

Reply:
that's nice, thanks for heads up!